Add process for Disconnected Server via Leapp (#1702)
theforeman/foreman-documentation: guides/common/modules/proc_upgrading-project-or-proxy-in-place-using-leapp.adoc

> [id="upgrading-{project-context}-or-proxy-in-place-using-leapp_{context}"]
> = Upgrading {Project} or {SmartProxy} to {EL} 8 In-Place Using Leapp

Reviewer: ```suggestion
[id="upgrading-{project-context}-server-or-{smart-proxy-context}-server-in-place-using-leapp_{context}"]
= Upgrading {ProjectServer} or {SmartProxyServer} to {EL} 8 In-Place Using Leapp
```

+
----
# curl -o /etc/yum.repos.d/theforeman-leapp.repo https://copr.fedorainfracloud.org/coprs/g/theforeman/leapp/repo/epel-7/group_theforeman-leapp-epel-7.repo
----
endif::[]
+
On {RHEL}, enable the `rhel-7-server-extras-rpms` repository:
+
----
# subscription-manager repos --enable=rhel-7-server-extras-rpms
----

. Install required packages:
[options="nowrap", subs="+quotes,verbatim,attributes"]
+
----
# {package-install-project} leapp leapp-repository
----
ifdef::satellite[]
. For Leapp to perform the upgrade in a disconnected environment, download the metadata and manually extract, as described in https://access.redhat.com/articles/3664871[Leapp utility metadata in-place upgrades of RHEL for disconnected upgrades].

. Set up the following repositories to perform the upgrade in a disconnected environment:
.. `/etc/yum.repos.d/rhel8.repo`:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
[BaseOS]
name={RepoRHEL8BaseOS}
baseurl=http://_server.example.com_/rhel8/BaseOS/

[AppStream]
name={RepoRHEL8AppStream}
baseurl=http://_server.example.com_/rhel8/AppStream/
----

> +

Reviewer: ```suggestion

```

.. `/etc/yum.repos.d/{project-context}.repo:`
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
[{RepoRHEL8ServerSatelliteServerProductVersion}]
name={RepoRHEL8ServerSatelliteServerProductVersion}
baseurl=http://_server.example.com_/sat6/Satellite/

[{RepoRHEL8ServerSatelliteMaintenanceProductVersion}]
name={RepoRHEL8ServerSatelliteMaintenanceProductVersion}
baseurl=http://_server.example.com_/sat6/Maintenance/
----
endif::[]

ifdef::foreman-el,katello[]
. Install additional OS specific packages (`leapp-data-almalinux` for AlmaLinux, `leapp-data-centos` for CentOS Stream, or `leapp-data-rocky` for Rocky Linux).
Note that this is not required for {RHEL} based installations.
+
----
# yum install leapp-data-centos
----

> +

Reviewer: ```suggestion
```

enabled=1
gpgcheck=1
module_hotfixes=1

[leapp-pulpcore]
name=pulpcore: Fetch, Upload, Organize, and Distribute Software Packages.
baseurl=https://yum.theforeman.org/pulpcore/{PulpcoreVersion}/el8/$basearch/
gpgkey=https://yum.theforeman.org/pulpcore/{PulpcoreVersion}/GPG-RPM-KEY-pulpcore
enabled=1
gpgcheck=1
module_hotfixes=1
endif::[]

[leapp-foreman-plugins]
name=Foreman plugins {ProjectVersion}
baseurl=https://yum.theforeman.org/plugins/{ProjectVersion}/el8/$basearch
enabled=1
gpgcheck=0
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman
module_hotfixes=1

[leapp-foreman-client]
name=Foreman client {ProjectVersion}
baseurl=https://yum.theforeman.org/client/{ProjectVersion}/el8/$basearch
enabled=1
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-foreman-client

[leapp-puppet7]
name=Puppet 7 Repository el 8 - $basearch
baseurl=http://yum.puppetlabs.com/puppet7/el/8/$basearch
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-puppet7-release
       file:///etc/pki/rpm-gpg/RPM-GPG-KEY-2025-04-06-puppet7-release
enabled=1
gpgcheck=1
----

* If you are using Puppet 6 instead of Puppet 7, replace the `7` with a `6` in the `leapp-puppet7` entry.

> * You need a Puppet repository for the Puppet agent that the installer is using.

Reviewer: Should this be in the Prerequisites perhaps?

> . We do not support {EL} 8 installations with EPEL 8 enabled, so remove `epel-release`:

Reviewer: ```suggestion
. {Project} does not support {EL} 8 installations with EPEL 8 enabled, so remove `epel-release`:
```

* "package rubygem-railties-6.0.4.7-1.el8.noarch requires rubygem(thor) < 2.0, but none of the providers can be installed"
--

+
If this happens, do the following to clean up packages that cannot automatically upgrade (`rubygem(thor)` and `rubygem(railties)` in the example above):

+
----
# yum remove rubygem-thor rubygem-railties
----
endif::[]

+
If `leapp preupgrade` inhibits the upgrade with *Unsupported network configuration* because there are multiple legacy named network interfaces, follow the instructions shown by Leapp to rename the interfaces, followed by an installer run to reconfigure {Project} or {SmartProxy} to use the new interface names:
+
[options="nowrap" subs="attributes"]
----
# {foreman-installer} --help |grep 'interface.*eth'
    --foreman-proxy-dhcp-interface  DHCP listen interface (current: "eth0")
    --foreman-proxy-dns-interface  DNS interface (current: "eth0")
----
+
If `eth0` was renamed to `em0`, call the installer to use the new interface name with:
+
[options="nowrap" subs="attributes"]
----
# {foreman-installer} --foreman-proxy-dhcp-interface=em0 --foreman-proxy-dns-interface=em0
----

. Ensure `leapp preupgrade` has no issues.

> . Run:

Reviewer: ```suggestion
. Perform the upgrade:
```

For more information, see xref:upgrading_a_disconnected_satellite[].
* For more information on customizing the Leapp upgrade for your environment, see https://access.redhat.com/articles/4977891[Customizing your {RHEL} in-place upgrade].
* Since Leapp completes part of the upgrade in a container that has no access to additional ISO mounts, the repositories cannot be served from a locally mounted ISO but must be delivered over the network from a different machine.
* For more information, see https://access.redhat.com/solutions/5492401[How to in-place upgrade an offline / disconnected RHEL 7 machine to RHEL 8 with Leapp?]
endif::[]
ifndef::satellite[]
* Access to available repositories or a local mirror of repositories.
endif::[]
* If you previously upgraded {Project} or {SmartProxy} from an earlier version, and the `/var/lib/pgsql` contained the PostgreSQL database content before the migration from PostgreSQL 9 to PostgreSQL 12 from the SCL, empty `/var/lib/pgsql` before proceeding.
* During the upgrade, the PostgreSQL data is moved from `/var/opt/rh/rh-postgresql12/lib/pgsql/data/` to `/var/lib/pgsql/data/`.

> If these two paths reside on the same partition, no further action is required.

Reviewer 2: As noted in https://github.com/theforeman/foreman-documentation/pull/1427#discussion_r989823230, those two are currently rendered under "disconnected":
![image](https://user-images.githubusercontent.com/94284/195817114-6d518b48-16b0-4bb5-8d4b-46e37cf238d7.png)
However, the PostgreSQL information applies equally to both connected and disconnected usescases.

# leapp upgrade
----

ifdef::satellite[]
+

> . If you run {Project} in a disconnected environment, add the `--no-rhsm` and `--enablerepo` parameters:

Reviewer 2: This should be additional information to the previous `leapp upgrade` step, not an own step.

+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# leapp upgrade \
--no-rhsm \
--enablerepo BaseOS \
--enablerepo AppStream \
--enablerepo {RepoRHEL8ServerSatelliteServerProductVersion} \
--enablerepo {RepoRHEL8ServerSatelliteMaintenanceProductVersion}
----

> 

Reviewer 3: ```suggestion
endif::[]

```
@evgeni you are indeed correct about the wrong alignment of the sections. I discovered this missing endif that was causing the issue.

Reviewer 2: Indeed, this `ifdef` is not properly closed, good find!

ifdef::satellite[]
. Complete the post-upgrade steps described in https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/upgrading_from_rhel_7_to_rhel_8/verifying-the-post-upgrade-state-of-the-rhel-8-system_upgrading-from-rhel-7-to-rhel-8[Verifying the post-upgrade state of the RHEL 8 system] in the _Upgrading from RHEL 7 to RHEL 8_ guide.
endif::[]

> . If you require SELinux to be in enforcing mode, run the following command before changing SELinux to enforcing mode:

Reviewer 2: This is a regression, the old text was correct, the new one isn't anymore.
